Source/Parallelization/WarpXRegrid.cpp

```diff
@@ -140,14 +144,61 @@ WarpX::LoadBalance ()
 }
 
 
+template <typename MultiFabType> void
+RemakeMultiFab (std::unique_ptr<MultiFabType>& mf, const DistributionMapping& dm,
+                const bool redistribute)
+{
+    if (mf == nullptr) return;
+    const IntVect& ng = mf->nGrowVect();
+    auto pmf = std::make_unique<MultiFabType>(mf->boxArray(), dm, mf->nComp(), ng);
+    if (redistribute) pmf->Redistribute(*mf, 0, 0, mf->nComp(), ng);
+    mf = std::move(pmf);
+}
+
 void
 WarpX::RemakeLevel (int lev, Real /*time*/, const BoxArray& ba, const DistributionMapping& dm)
 {
     if (ba == boxArray(lev))
     {
         if (ParallelDescriptor::NProcs() == 1) return;
 
+        // Fine patch
+        for (int idim=0; idim < 3; ++idim)
+        {
+            RemakeMultiFab(Bfield_fp[lev][idim], dm, true);
+            RemakeMultiFab(Efield_fp[lev][idim], dm, true);
+            RemakeMultiFab(current_fp[lev][idim], dm, false);
+            RemakeMultiFab(current_store[lev][idim], dm, false);
+
+#ifdef AMREX_USE_EB
+            if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::Yee ||
+                WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT ||
+                WarpX::maxwell_solver_id == MaxwellSolverAlgo::CKC){
+                RemakeMultiFab(m_edge_lengths[lev][idim], dm, false);
+                RemakeMultiFab(m_face_areas[lev][idim], dm, false);
+                if(WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT){
+                    RemakeMultiFab(Venl[lev][idim], dm, false);
+                    RemakeMultiFab(m_flag_info_face[lev][idim], dm, false);
+                    RemakeMultiFab(m_flag_ext_face[lev][idim], dm, false);
+                    RemakeMultiFab(m_area_mod[lev][idim], dm, false);
+                    RemakeMultiFab(ECTRhofield[lev][idim], dm, false);
+                }
+
+                m_borrowing[lev][idim] = std::make_unique<amrex::LayoutData<FaceInfoBox>>(amrex::convert(ba, Bfield_fp[lev][idim]->ixType().toIntVect()), dm);
```

Comment:
@WeiqunZhang or @lgiacome could you guys please confirm that this is the correct way to "reset" `m_borrowing`?

Reply:
Also, should this go inside the `if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT)` block? Plus I still need to add the `ComputeFaceExtensions()` call, right?

Reply:
Sorry, I said I would look into it but then I ran out of time...
So the reset looks fine to me.

After that outside the loop over idim you need to call first `MarkCells()` and after you call `ComputeFaceExtensions()`.

This has to happen only `if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT)`

I can add the suggestion so that you can just commit it.

Reply:
Sounds good. I'll wait for the suggestion. Thanks!

Reply:
I should have added the suggestions. It should be correct, but it would be great if you could have a second look at the style 😅

Reply:
Thanks a lot! I think this PR should now be just about ready to go.